### ISSUE TITLE
Breaking Bootstrap 5 Floating Labels

### ISSUE THREAD

**Issue by Silvenga** [parcel-bundler/lightningcss#577]
Bootstrap has the following CSS (after transpiling):

```css
.form-floating>.form-control:focus,
.form-floating>.form-control:not(:placeholder-shown),
.form-floating>.form-control-plaintext:focus,
.form-floating>.form-control-plaintext:not(:placeholder-shown),
:-webkit-any(.form-floating > .form-control:-webkit-autofill, .form-floating > .form-control-plaintext:-webkit-autofill),
.form-floating>.form-select {
  padding-top: 1.625rem;
  padding-bottom: .625rem;
}

.form-floating>.form-control:focus,
.form-floating>.form-control:not(:placeholder-shown),
.form-floating>.form-control-plaintext:focus,
.form-floating>.form-control-plaintext:not(:placeholder-shown),
:is(.form-floating > .form-control:autofill, .form-floating > .form-control-plaintext:autofill),
.form-floating>.form-select {
  padding-top: 1.625rem;
  padding-bottom: .625rem;
}
```

Which originates from the source: 

https://github.com/twbs/bootstrap/blob/fee563d1fca2acea0bde65ab382b4b79f93f600d/scss/forms/_floating-labels.scss#L73-L79

This ultimately produces the following (I formatted the resulting CSS for readability):

```css
.form-floating>.form-control:focus,
.form-floating>.form-control:not(:placeholder-shown),
.form-floating>.form-control-plaintext:focus,
.form-floating>.form-control-plaintext:not(:placeholder-shown),
:-webkit-any(.form-floating>.form-control:-webkit-autofill, .form-floating>.form-control-plaintext:-webkit-autofill),
.form-floating>.form-select,
.form-floating>.form-control:focus,
.form-floating>.form-control:not(:placeholder-shown),
.form-floating>.form-control-plaintext:focus,
.form-floating>.form-control-plaintext:not(:placeholder-shown),
:-webkit-any(.form-floating>.form-control:-webkit-autofill, .form-floating>.form-control-plaintext:-webkit-autofill),
.form-floating>.form-select {
    padding-top: 1.625rem;
    padding-bottom: .625rem
}

.form-floating>.form-control:focus,
.form-floating>.form-control:not(:placeholder-shown),
.form-floating>.form-control-plaintext:focus,
.form-floating>.form-control-plaintext:not(:placeholder-shown),
:-webkit-any(.form-floating>.form-control:autofill, .form-floating>.form-control-plaintext:autofill),
.form-floating>.form-select,
.form-floating>.form-control:focus,
.form-floating>.form-control:not(:placeholder-shown),
.form-floating>.form-control-plaintext:focus,
.form-floating>.form-control-plaintext:not(:placeholder-shown),
:is(.form-floating>.form-control:autofill, .form-floating>.form-control-plaintext:autofill),
.form-floating>.form-select {
    padding-top: 1.625rem;
    padding-bottom: .625rem
}
```

I believe the problem originates in the second block:

```css
.form-floating>.form-control:focus,
.form-floating>.form-control:not(:placeholder-shown),
.form-floating>.form-control-plaintext:focus,
.form-floating>.form-control-plaintext:not(:placeholder-shown),
:-webkit-any(.form-floating>.form-control:autofill, .form-floating>.form-control-plaintext:autofill),
.form-floating>.form-select,
.form-floating>.form-control:focus,
.form-floating>.form-control:not(:placeholder-shown),
.form-floating>.form-control-plaintext:focus,
.form-floating>.form-control-plaintext:not(:placeholder-shown),
:is(.form-floating>.form-control:autofill, .form-floating>.form-control-plaintext:autofill),
.form-floating>.form-select {
    padding-top: 1.625rem;
    padding-bottom: .625rem
}
```

Note how both `-webkit-any` and `is` are included in the same group (arguably duplicated, but that's a separate "issue"). The rending problem doesn't occur when the blocks are separate (as shown in the original Bootstrap SASS). Bootstrap's SASS notes that that the reason they duplicated their selector was to fix the problem that LightningCSS is now causing:

> Duplicated because `:-webkit-autofill` invalidates other selectors when grouped

So it appears that LightningCSS is ultimately causing floating labels to break when optimizations are enabled.

[Playground link](https://lightningcss.dev/playground/index.html#%7B%22minify%22%3Atrue%2C%22nesting%22%3Atrue%2C%22customMedia%22%3Atrue%2C%22cssModules%22%3Afalse%2C%22analyzeDependencies%22%3Afalse%2C%22targets%22%3A%7B%22chrome%22%3A6225920%7D%2C%22include%22%3A0%2C%22exclude%22%3A0%2C%22source%22%3A%22.form-floating%3E.form-control%3Afocus%2C%5Cn.form-floating%3E.form-control%3Anot(%3Aplaceholder-shown)%2C%5Cn.form-floating%3E.form-control-plaintext%3Afocus%2C%5Cn.form-floating%3E.form-control-plaintext%3Anot(%3Aplaceholder-shown)%2C%5Cn%3A-webkit-any(.form-floating%20%3E%20.form-control%3A-webkit-autofill%2C%20.form-floating%20%3E%20.form-control-plaintext%3A-webkit-autofill)%2C%5Cn.form-floating%3E.form-select%20%7B%5Cn%20%20padding-top%3A%201.625rem%3B%5Cn%20%20padding-bottom%3A%20.625rem%3B%5Cn%7D%5Cn%5Cn.form-floating%3E.form-control%3Afocus%2C%5Cn.form-floating%3E.form-control%3Anot(%3Aplaceholder-shown)%2C%5Cn.form-floating%3E.form-control-plaintext%3Afocus%2C%5Cn.form-floating%3E.form-control-plaintext%3Anot(%3Aplaceholder-shown)%2C%5Cn%3Ais(.form-floating%20%3E%20.form-control%3Aautofill%2C%20.form-floating%20%3E%20.form-control-plaintext%3Aautofill)%2C%5Cn.form-floating%3E.form-select%20%7B%5Cn%20%20padding-top%3A%201.625rem%3B%5Cn%20%20padding-bottom%3A%20.625rem%3B%5Cn%7D%5Cn%22%2C%22visitorEnabled%22%3Afalse%2C%22visitor%22%3A%22%7B%5Cn%20%20Color(color)%20%7B%5Cn%20%20%20%20if%20(color.type%20%3D%3D%3D%20'rgb')%20%7B%5Cn%20%20%20%20%20%20color.g%20%3D%200%3B%5Cn%20%20%20%20%20%20return%20color%3B%5Cn%20%20%20%20%7D%5Cn%20%20%7D%5Cn%7D%22%2C%22unusedSymbols%22%3A%5B%5D%2C%22version%22%3A%22local%22%7D)

**Comment by devongovett** [comment]
Fixed as part of f782754275248b7abf4cf316c6ad684000d6e2df

**Comment by Silvenga** [comment]
I can confirm `lightningcss@1.22.0` fixes this issue.

`package.json`:
```json
  "resolutions": {
    "lightningcss": "^1.22.0"
  }
```

Thanks @devongovett!
